crm-project/src/main/java/CRM/service/UserService.java

```diff
@@ -149,28 +151,54 @@ public List<User> getAllInBoard(long boardId) throws AccountNotFoundException {
+    public List<User> updateUserToBoard(long userId, long boardId, long permissionId) throws AccountNotFoundException {
         User user;
         Board board;
         try {
             user = Validations.doesIdExists(userId, userRepository);
             board = Validations.doesIdExists(boardId, boardRepository);
-
         } catch (NoSuchElementException e) {
             throw new AccountNotFoundException(ExceptionMessage.ACCOUNT_DOES_NOT_EXISTS.toString());
         }
 
-        NotificationSetting notificationSetting = Validations.doesIdExists(2L, settingRepository);
+        if(user.equals(board.getCreatorUser())){
+            throw new IllegalArgumentException(ExceptionMessage.ADMIN_CANT_CHANGE_HIS_PERMISSION.toString());
+        }
 
-        UserPermission userPermission = new UserPermission();
-        userPermission.setId(0L);
-        userPermission.setUser(user);
-        userPermission.setPermission(Permission.USER);
+        UserPermission userPermissionInBoard = null;
+        Set<UserPermission> userPermissionsSet = board.getUsersPermissions();
 
-        board.addUserPermissionToBoard(userPermission);
+        for (UserPermission userInBoard : userPermissionsSet) {
+            if (userInBoard.getId().equals(userId)) {
+                userPermissionInBoard = userInBoard;
+                break;
+            }
+        }
 
-        createDefaultSettingForNewUserInBoard(user, board, notificationSetting);
+        Permission permissionRequest = Permission.values()[(int) permissionId];
+
+        if (userPermissionInBoard == null) { //create new
+            if(!permissionRequest.equals(Permission.ADMIN) && !permissionRequest.equals(Permission.UNAUTHORIZED)) createNewUserPermission(user, permissionRequest, board);
+            else throw new IllegalArgumentException(ExceptionMessage.CANT_ASSIGN_PERMISSION.toString());
+        } else if (permissionRequest.equals(Permission.ADMIN)) {
+            throw new IllegalArgumentException(ExceptionMessage.PERMISSION_NOT_ALLOWED.toString());
+        } else if (permissionRequest.equals(Permission.USER) && !userPermissionInBoard.getPermission().equals(Permission.USER)) {
+            userPermissionInBoard.setPermission(permissionRequest);
+        } else if (permissionRequest.equals(Permission.LEADER) && !userPermissionInBoard.getPermission().equals(Permission.LEADER)) {
+            userPermissionInBoard.setPermission(permissionRequest);
+        } else if (permissionRequest.equals(Permission.UNAUTHORIZED)) {
+            userPermissionsSet.remove(userPermissionInBoard);
+        } else {
+            throw new IllegalArgumentException(ExceptionMessage.USER_ALREADY_HAS_THIS_PERMISSION.toString());
+        }
+
+        List<User> users = new ArrayList<>();
+        users.add(board.getCreatorUser());
+        for (UserPermission addUSer: userPermissionsSet) {
+            users.add(addUSer.getUser());
+        }
         boardRepository.save(board);
+        return users;
     }
```

Comment:
Also this for loop.
Basically, everything the board can do by itself, do it inside the board class and keep the controller clean